### ISSUE TITLE
Suicidal AI behavior: bots like to swim into propellers, get stuck, and get killed

### ISSUE THREAD

**Issue by Luckspeare** [Regalis11/Barotrauma#7969]
In SP, if you're controlling an AI character swimming somewhere to the rear of the sub, and you switch back to, say, the captain to move the sub a bit, then switch back to the AI character, you're liable to find he swap right into the screws/propeller/engine and is now stuck and dying.  They probably should steer clear of lethal hazards.

**Comment by Luckspeare** [comment]
I should note this isn't at all rare -- since AI are finding the shortest path to where they need to go, if that involves going around the rear of the ship, they're going to eat screw.

**Comment by itchyOwl** [comment]
Does it happen on vanilla subs or custom subs? There's no custom logic for the bots to avoid the propellers, so it the path goes too close to the propeller, they can get injured by it. On vanilla subs, the path shouldn't be that close to the propeller. If the waypoint generator adds waypoints too close to the propeller, I think we might need to adjust that. But if the path is ok, and the bots still often navigate to the propeller, there's something wrong in the navigation logic.

So, in other words: is it an issue with the waypoint generator or the navigation logic?

**Comment by MaaiG** [comment]
> If the waypoint generator adds waypoints too close to the propeller, I think we might need to adjust that.

https://user-images.githubusercontent.com/93348499/147946441-6bec1206-8abd-4319-ad46-634253630100.mp4

**Comment by Luckspeare** [comment]
This was indeed on a mod sub.  I wasn't even aware that pathing was necessary outside the ship.   In this case, though, I am not sure pathing was the issue -- I had an AI crewmember on follow, and he got stuck in the propeller trying to reach me.

**Comment by Rokvach** [comment]
No issues, closing.
